### PR TITLE
New UI: Show new left sidebar and hide old sidebar

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
None

**Description**
Update left nav to use new style sidebar under feature flag

image

### CODE REVIEW

File: packages/studio-base/src/Workspace.tsx [hunk 1]
```diff
@@ -521,6 +521,12 @@ export default function Workspace(props: WorkspaceProps): JSX.Element {
     };
   }, [enableNewTopNav]);
 
+  const TopicListSidebar = useMemo(() => {
+    return function TopicListSidebarImpl() {
+      return <TopicList onSelectDataSourceAction={() => setShowOpenDialog({ view: "start" })} />;
+    };
+  }, []);
```
Comment (jtbandes) [inline packages/studio-base/src/Workspace.tsx:528]: might be a little cleaner as useCallback rather than useMemo

Comment (2metres) [reply packages/studio-base/src/Workspace.tsx:528]: We have a mixture of `useMemo` and `useCallback` here. Should I prefer one over the other?

Comment (jtbandes) [reply packages/studio-base/src/Workspace.tsx:528]: Usually useCallback when the thing is a function and useMemo otherwise

**Review by jtbandes — CHANGES_REQUESTED** [review]
Spacing adjustments make storybook changes hard to review :(
image

These stories look wrong:
image

**Comment by 2metres** [comment]
> These stories look wrong: image

What looks wrong about them? Should we add test for the new UI?

**Comment by amacneil** [comment]
Panel settings does not open when I click a panel's cog icon.

**Review by jtbandes — APPROVED** [review]
Code changes looking good but a couple of the stories are missing their sidebar.

File: packages/studio-base/src/components/Sidebars/NewSidebar.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  label,
+  defaultLeftKey,
+  defaultRightKey,
+  enableAppBar,
+  height,
+}: {
+  label?: string;
+  defaultLeftKey?: string | undefined;
+  defaultRightKey?: string | undefined;
+  enableAppBar?: boolean;
```
Comment (jtbandes) [inline packages/studio-base/src/components/Sidebars/NewSidebar.stories.tsx:70]: Looks like this prop is unused now

File: packages/studio-base/src/components/Sidebars/NewSidebar.stories.tsx [hunk 3]
```diff
@@ -133,7 +127,7 @@ LeftClosed.play = async () => {
 };
 
 // Right
-export const RightOpen = (): JSX.Element => <Story defaultRightKey="x" />;
+export const RightOpen = (): JSX.Element => <Story />;
```
Comment (jtbandes) [inline packages/studio-base/src/components/Sidebars/NewSidebar.stories.tsx:130]: Is this story doing anything at all?

**Review by jtbandes — APPROVED** [review]

**Comment by amacneil** [comment]
Product review feedback:
- Remove left sidebar "open data source" button for now
- Right hand sidebar needs to be wider by default

Follow up meeting about panel settings position

**Pull request merged**